packages/core/src/middleware/koa-camelcase-transform.test.ts

```diff
@@ -0,0 +1,42 @@
+import { createMockContext } from '@shopify/jest-koa-mocks';
+
+import koaCamelcaseTransform, { ResponseKeyCase } from './koa-camelcase-transform';
+
+const next = jest.fn();
+
+describe('koa-camelcase-transform', () => {
+  it('should transform body snakecase to camel case with keycase header camecase', async () => {
+    const ctx = createMockContext({
+      headers: {
+        'response-keycase': ResponseKeyCase.camelcase,
+      },
+    });
+
+    ctx.body = { redirect_uri: 'foo' };
+
+    await koaCamelcaseTransform()(ctx, next);
+
+    expect(ctx.body).toHaveProperty('redirectUri');
+  });
+
+  it('should not transform body to camel case without keycase header', async () => {
+    const ctx = createMockContext();
+    ctx.body = { redirect_uri: 'foo' };
+    await koaCamelcaseTransform()(ctx, next);
+
+    expect(ctx.body).toHaveProperty('redirect_uri');
```

Comment:
should be `redirectUri`?

Reply:
please check the test name, in no key case header provider, shouldn't apply the transformation.

Reply:
i thought key names default to camelcase.

after re-reading [discussions](
https://silverhand-io.slack.com/archives/C02A8G4HVAM/p1642410681104400?thread_ts=1642408934.096700&channel=C02A8G4HVAM&message_ts=1642410681.104400), i know what you mean now.